signinapp/model.py

```diff
@@ -164,13 +188,57 @@ def make(name: str, password: str, role: Role, preferred_name: str = None,
                     subteam=subteam,
                     approved=approved)
 
+    @staticmethod
+    def make_guardian(student: User, name: str, phone_number: str, email: str, preference: int = 1):
+        guardian_limted = Role.from_name("guardian_limited")
+        guardian = User(name=name,
+                        code=mk_hash(name),
+                        role_id=guardian_limted.id,
+                        phone_number=normalize_phone_number_for_storage(
+                            phone_number),
+                        email=email
+                        )
+        db.session.add(guardian)
+        db.session.flush()
+        guardian.guardian = Guardian.add_guardian(
+            student=student, parent_id=guardian.id, preference=preference)
+        return guardian
+
     @ staticmethod
     def get_canonical(name) -> User | None:
         ' Look up user by name '
         code = mk_hash(name)
         return User.query.filter_by(code=code).one_or_none()
 
 
+class Guardian(db.Model):
+    """
+    This table is a bit strange as it has a one to one link with a User (Parent) as well as
+    Many to Many links with User (Children).
+
+    """
+    __tablename__ = "guardians"
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"))
+    preference = db.Column(db.Integer)
+
+    # One to One: Links User to row in Guardian table
+    user = relationship("User", back_populates="guardian")
+
+    # Many to Many: Links Guardian to Children
+    students = relationship(
+        "User", secondary=parent_child_association_table, back_populates="guardians")
+
+    @classmethod
+    def add_guardian(cls, student: User, parent_id: int, preference: int = None) -> Guardian:
+        guardian = Guardian(user_id=parent_id,
+                            preference=preference, students=[student])
+        db.session.add(guardian)
+
+    def update_guardian(self, student: User):
```

Comment:
This doesn't update anything about the guardian itself?

Reply:
Shh naming is hard. `add_student`?

Reply:
`add_to_student`? Or a student function `add_guardian`?